### PR TITLE
Use StatusCodes constants instead of literals in the ProducesResponseType code fix

### PR DESCRIPTION
**Pull request by khellang**
I'm a fan of using the constants. What do you think? @pranavkm?

### CODE REVIEW

File: test/Mvc.Api.Analyzers.Test/TestFiles/AddResponseTypeAttributeCodeFixProviderIntegrationTest/CodeFixAddsMissingStatusCodes.Output.cs [hunk 1]
```diff
@@ -5,8 +5,8 @@
     public class CodeFixAddsMissingStatusCodes : ControllerBase
     {
         [ProducesResponseType(404)]
-        [ProducesResponseType(200)]
-        [ProducesResponseType(400)]
+        [ProducesResponseType(Http.StatusCodes.Status200OK)]
```
Comment (pranavkm) [inline test/Mvc.Api.Analyzers.Test/TestFiles/AddResponseTypeAttributeCodeFixProviderIntegrationTest/CodeFixAddsMissingStatusCodes.Output.cs]: Would it make sense to try adding this as a `using`? In the most common case, controllers have a reference to `Microsoft.AspNetCore.Mvc` whereas `Microsoft.AspNetCore.Http` is a rarity.

Comment (khellang) [reply test/Mvc.Api.Analyzers.Test/TestFiles/AddResponseTypeAttributeCodeFixProviderIntegrationTest/CodeFixAddsMissingStatusCodes.Output.cs]: Yeah, the thought did cross my mind. Your call 😊

**Comment by khellang** [comment]
BTW, I have a PR lined up adding support for

 - `return StatusCode(...)`
 - `return new StatusCodeResult(...)`
 - `return new ObjectResult(...) { StatusCode = ... }`

as well, if you're interested. It builds on top of this branch, so I'll submit it once this is merged/rejected.

File: test/Mvc.Api.Analyzers.Test/TestFiles/AddResponseTypeAttributeCodeFixProviderIntegrationTest/CodeFixAddsMissingStatusCodes.Output.cs (hunk 1 above)
Comment (pranavkm) [reply test/Mvc.Api.Analyzers.Test/TestFiles/AddResponseTypeAttributeCodeFixProviderIntegrationTest/CodeFixAddsMissingStatusCodes.Output.cs]: Let's go with the `using`. The results would look a bit cleaner.

**Comment by khellang** [comment]
Using directives added ✅

File: src/Microsoft.AspNetCore.Mvc.Api.Analyzers/AddResponseTypeAttributeCodeFixAction.cs [hunk 2]
```diff
@@ -64,7 +67,16 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
...
             }
 
-            return documentEditor.GetChangedDocument();
+            var document = documentEditor.GetChangedDocument();
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken);
+
+            if (root is CompilationUnitSyntax compilationUnit && addUsingDirective)
+            {
+                root = compilationUnit.AddUsings(SyntaxFactory.UsingDirective(SyntaxFactory.ParseName("Microsoft.AspNetCore.Http")));
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Api.Analyzers/AddResponseTypeAttributeCodeFixAction.cs]: Does this result in a compile error if the using already exists? Could you add a test that covers this?

File: src/Microsoft.AspNetCore.Mvc.Api.Analyzers/AddResponseTypeAttributeCodeFixAction.cs [hunk 3]
```diff
@@ -64,7 +67,16 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
                 documentEditor.RemoveNode(attributeSyntax);
             }
 
-            return documentEditor.GetChangedDocument();
+            var document = documentEditor.GetChangedDocument();
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken);
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Api.Analyzers/AddResponseTypeAttributeCodeFixAction.cs]: `.ConfigureAwait(false);`

File: src/Microsoft.AspNetCore.Mvc.Api.Analyzers/AddResponseTypeAttributeCodeFixAction.cs (hunk 2 above)
Comment (khellang) [reply src/Microsoft.AspNetCore.Mvc.Api.Analyzers/AddResponseTypeAttributeCodeFixAction.cs]: Nice catch!

> The using directive for 'Microsoft.AspNetCore.Http' appeared previously in this namespace

**Comment by pranavkm** [comment]
Thanks. I'll merge this in once the build succeeds.

**Comment by pranavkm** [comment]
Thanks for the PR!

**Pull request merged**